src/components/legend.ts

```diff
@@ -1,4 +1,233 @@
 namespace Plottable.Components {
+  /**
+   * The Legend's column representation. Stores position information
+   * as well as data
+   */
+  class LegendColumn<T> {
```

Comment:
This seems like a struct, and since you're not exporting it, why not just make it one instead of all the extra getter setter mess?